packages/shared/lib/core/account/actions/tryEditSelectedAccountPersistedData.ts

```diff
@@ -1,17 +1,22 @@
 import { localize } from '@core/i18n'
-import { isActiveLedgerProfile, updateActiveAccountMetadata } from '@core/profile'
+import { isActiveLedgerProfile, updateActiveAccountPersistedData } from '@core/profile'
 import { displayNotificationForLedgerProfile } from '@core/ledger'
 import { showAppNotification } from '@auxiliary/notification'
 import { get } from 'svelte/store'
-import { IAccountMetadata } from '../interfaces'
+import { IAccountPersistedData } from '../interfaces'
 import { selectedAccount, updateSelectedAccount } from '../stores'
 
-export async function tryEditSelectedAccountMetadata(metadata: Partial<IAccountMetadata>): Promise<void> {
+export async function tryEditSelectedAccountPersistedData(
+    partialAccountPersistedData: Partial<IAccountPersistedData>
+): Promise<void> {
     try {
         // TODO: Replace Promise.resolve() with update metadata in wallet.rs when api is exposed
         await Promise.resolve()
-        updateActiveAccountMetadata(get(selectedAccount)?.index, metadata)
-        updateSelectedAccount(metadata)
+        const _selectedAccount = get(selectedAccount)
+        if (_selectedAccount) {
+            updateActiveAccountPersistedData(_selectedAccount.index, partialAccountPersistedData)
+        }
+        updateSelectedAccount(partialAccountPersistedData)
```

Comment:
you can move this line into the above `if` as well because `updateSelectedAccount` doesnt make sense without `_selectedAccount`

Reply:
updated